builder/docker/communicator.go

```diff
@@ -310,3 +319,70 @@ func (c *Communicator) run(cmd *exec.Cmd, remote *packer.RemoteCmd, stdin io.Wri
 	// Set the exit status which triggers waiters
 	remote.SetExited(exitStatus)
 }
+
+// TODO Workaround for #5307. Remove once #5409 is fixed.
+func (c *Communicator) fixDestinationOwner(destination string) error {
+	if c.containerUser == nil {
+		containerUser, err := c.discoverContainerUser()
+		if err != nil {
+			return err
+		}
+		c.containerUser = &containerUser
+	}
+
+	if *c.containerUser != "" {
+		chownArgs := []string{
```

Comment:
all of this I think can be replaced by 

```
cmd = &packer.RemoteCmd{
	Command: fmt.Sprintf("chown -R %s %s", *c.containerUser, destination)
}
return c.Start(cmd)
```

same with `discoverContainerUser`. `runLocalCommand` should be removed

Reply:
1. `chown` needs to be run as `root` and I believe that remote commands are executed as the user the container is running as or as `exec_user`

2. AFAIK, remote commands run inside the container and `discoverContainerUser` calls `docker inspect` so it must be executed on host, the only way I can think of about re-using remote command to discover user is by running `id -u` inside the container but (1) for me `docker inspect` seems more "kosher" and (2) if `exec_user` is used `id -u` would return that user which also might be unexpected.

Reply:
Yeah, you're right. In that case, I think runlocalcommand can be replaced with `exec.Command` and `cmd.CombinedOutput()`, since I don't think we care about distinguishing between stderr and stdout here.